### DISCUSSION TITLE
Add new Timesheet Meta won't save.

### DISCUSSION THREAD

**Discussion by YohannesGetu** [kimai/kimai#4243] [Questions & Answers]
when adding new Timesheet Meta field on CalculatorInterface implementation class by `
$externalId = (new TimesheetMeta())->setName('externalID')->setValue(1);
$record->setMetaField($externalId);` the meta is not saved all other changes on timesheet object is saved.

**Comment by kevinpapst** [comment]
This code should work.

**Reply by YohannesGetu** [reply]
i use this in CalculatorInterface implementation and it's not working, here is the code,

`<?php



namespace KimaiPlugin\WorkingShiftBundle\Timesheet\Calculator;

use App\Configuration\SystemConfiguration;
use App\Constants;
use App\Entity\TimesheetMeta;
use App\Entity\Timesheet;
use KimaiPlugin\WorkingShiftBundle\Repository\WorkingShiftRepository;
use KimaiPlugin\WorkingShiftBundle\Entity\WorkingShift;
use App\Timesheet\Calculator\RateCalculator;
use App\Timesheet\CalculatorInterface;
use Doctrine\ORM\PersistentCollection;

final class WorkingShiftRateCalculator implements CalculatorInterface
{
    private $calculator;
    private $systemConfiguration;
    private $workingShiftRepository;

    public function __construct(RateCalculator $calculator, SystemConfiguration $systemConfiguration, WorkingShiftRepository $workingShiftRepository)
    {
        $this->calculator = $calculator;
        $this->systemConfiguration = $systemConfiguration;
        $this->workingShiftRepository = $workingShiftRepository;
    }

    /**
     * @param Timesheet $record
     * @return void
     */
    public function calculate(Timesheet $record,  array $changeset): void
    {

        $workingShifts = $this->workingShiftRepository->findAll();

        if($workingShifts == null || sizeof($workingShifts) < 1) 
        {
            
            $record->setDescription($record->getDescription() . " ##With no working shift");
            
        } else{

            $workingShift= (new TimesheetMeta())->setName('shift')->setValue(sizeof($workingShifts));
           $record->setMetaField($workingShift);

        }

        
       

    }

    public function getPriority(): int
    {
        return 350;
    }
}`
